// speech_to_text_api_v30/src/model/EndpointProperties.js
/*
 * Speech to Text API v3.0
 * Speech to Text API v3.0.
 *
 * OpenAPI spec version: v3.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.21
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/EntityError'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./EntityError'));
  } else {
    // Browser globals (root is window)
    if (!root.SpeechToTextApiV30) {
      root.SpeechToTextApiV30 = {};
    }
    root.SpeechToTextApiV30.EndpointProperties = factory(root.SpeechToTextApiV30.ApiClient, root.SpeechToTextApiV30.EntityError);
  }
}(this, function(ApiClient, EntityError) {
  'use strict';

  /**
   * The EndpointProperties model module.
   * @module model/EndpointProperties
   * @version v3.0
   */

  /**
   * Constructs a new <code>EndpointProperties</code>.
   * @alias module:model/EndpointProperties
   * @class
   */
  var exports = function() {
  };

  /**
   * Constructs a <code>EndpointProperties</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/EndpointProperties} obj Optional instance to populate.
   * @return {module:model/EndpointProperties} The populated <code>EndpointProperties</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('loggingEnabled'))
        obj.loggingEnabled = ApiClient.convertToType(data['loggingEnabled'], 'Boolean');
      if (data.hasOwnProperty('timeToLive'))
        obj.timeToLive = ApiClient.convertToType(data['timeToLive'], 'String');
      if (data.hasOwnProperty('email'))
        obj.email = ApiClient.convertToType(data['email'], 'String');
      if (data.hasOwnProperty('error'))
        obj.error = EntityError.constructFromObject(data['error']);
    }
    return obj;
  }

  /**
   * A value indicating whether content logging (audio &amp; transcriptions) is being used for a deployment.
   * @member {Boolean} loggingEnabled
   */
  exports.prototype.loggingEnabled = undefined;

  /**
   * How long the endpoint will be kept in the system. Once the endpoint reaches the time to live  after completion (successful or failed) it will be automatically deleted. Not setting this value or setting  to 0 will disable automatic deletion.  The duration is encoded as ISO 8601 duration (\"PnYnMnDTnHnMnS\", see https://en.wikipedia.org/wiki/ISO_8601#Durations).
   * @member {String} timeToLive
   */
  exports.prototype.timeToLive = undefined;

  /**
   * The email address to send email notifications to in case the operation completes.  The value will be removed after successfully sending the email.
   * @member {String} email
   */
  exports.prototype.email = undefined;

  /**
   * The details of the error in case the entity is in a failed state.
   * @member {module:model/EntityError} error
   */
  exports.prototype.error = undefined;


  return exports;

}));
