// speech_to_text_api_v30/src/model/File.js
/*
 * Speech to Text API v3.0
 * Speech to Text API v3.0.
 *
 * OpenAPI spec version: v3.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.21
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/FileLinks', 'model/FileProperties'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./FileLinks'), require('./FileProperties'));
  } else {
    // Browser globals (root is window)
    if (!root.SpeechToTextApiV30) {
      root.SpeechToTextApiV30 = {};
    }
    root.SpeechToTextApiV30.File = factory(root.SpeechToTextApiV30.ApiClient, root.SpeechToTextApiV30.FileLinks, root.SpeechToTextApiV30.FileProperties);
  }
}(this, function(ApiClient, FileLinks, FileProperties) {
  'use strict';

  /**
   * The File model module.
   * @module model/File
   * @version v3.0
   */

  /**
   * Constructs a new <code>File</code>.
   * @alias module:model/File
   * @class
   */
  var exports = function() {
  };

  /**
   * Constructs a <code>File</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/File} obj Optional instance to populate.
   * @return {module:model/File} The populated <code>File</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('kind'))
        obj.kind = ApiClient.convertToType(data['kind'], 'String');
      if (data.hasOwnProperty('links'))
        obj.links = FileLinks.constructFromObject(data['links']);
      if (data.hasOwnProperty('createdDateTime'))
        obj.createdDateTime = ApiClient.convertToType(data['createdDateTime'], 'Date');
      if (data.hasOwnProperty('properties'))
        obj.properties = FileProperties.constructFromObject(data['properties']);
      if (data.hasOwnProperty('name'))
        obj.name = ApiClient.convertToType(data['name'], 'String');
      if (data.hasOwnProperty('self'))
        obj.self = ApiClient.convertToType(data['self'], 'String');
    }
    return obj;
  }

  /**
   * The kind of this file.
   * @member {module:model/File.KindEnum} kind
   */
  exports.prototype.kind = undefined;

  /**
   * The links for additional actions or content related to this file.
   * @member {module:model/FileLinks} links
   */
  exports.prototype.links = undefined;

  /**
   * The creation time of this file.  The time stamp is encoded as ISO 8601 date and time format  (see https://en.wikipedia.org/wiki/ISO_8601#Combined_date_and_time_representations).
   * @member {Date} createdDateTime
   */
  exports.prototype.createdDateTime = undefined;

  /**
   * Additional metadata provided by the service.
   * @member {module:model/FileProperties} properties
   */
  exports.prototype.properties = undefined;

  /**
   * The name of this file.
   * @member {String} name
   */
  exports.prototype.name = undefined;

  /**
   * The location of this entity.
   * @member {String} self
   */
  exports.prototype.self = undefined;



  /**
   * Allowed values for the <code>kind</code> property.
   * @enum {String}
   * @readonly
   */
  exports.KindEnum = {
    /**
     * value: "DatasetReport"
     * @const
     */
    datasetReport: "DatasetReport",

    /**
     * value: "Audio"
     * @const
     */
    audio: "Audio",

    /**
     * value: "LanguageData"
     * @const
     */
    languageData: "LanguageData",

    /**
     * value: "PronunciationData"
     * @const
     */
    pronunciationData: "PronunciationData",

    /**
     * value: "AcousticDataArchive"
     * @const
     */
    acousticDataArchive: "AcousticDataArchive",

    /**
     * value: "AcousticDataTranscriptionV2"
     * @const
     */
    acousticDataTranscriptionV2: "AcousticDataTranscriptionV2",

    /**
     * value: "Transcription"
     * @const
     */
    transcription: "Transcription",

    /**
     * value: "TranscriptionReport"
     * @const
     */
    transcriptionReport: "TranscriptionReport",

    /**
     * value: "EvaluationDetails"
     * @const
     */
    evaluationDetails: "EvaluationDetails"
  };

  return exports;

}));
