// speech_to_text_api_v30/src/model/ModelManifest.js
/*
 * Speech to Text API v3.0
 * Speech to Text API v3.0.
 *
 * OpenAPI spec version: v3.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.21
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/EntityReference', 'model/ModelFile'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./EntityReference'), require('./ModelFile'));
  } else {
    // Browser globals (root is window)
    if (!root.SpeechToTextApiV30) {
      root.SpeechToTextApiV30 = {};
    }
    root.SpeechToTextApiV30.ModelManifest = factory(root.SpeechToTextApiV30.ApiClient, root.SpeechToTextApiV30.EntityReference, root.SpeechToTextApiV30.ModelFile);
  }
}(this, function(ApiClient, EntityReference, ModelFile) {
  'use strict';

  /**
   * The ModelManifest model module.
   * @module model/ModelManifest
   * @version v3.0
   */

  /**
   * Constructs a new <code>ModelManifest</code>.
   * @alias module:model/ModelManifest
   * @class
   */
  var exports = function() {
  };

  /**
   * Constructs a <code>ModelManifest</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/ModelManifest} obj Optional instance to populate.
   * @return {module:model/ModelManifest} The populated <code>ModelManifest</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('model'))
        obj.model = EntityReference.constructFromObject(data['model']);
      if (data.hasOwnProperty('modelFiles'))
        obj.modelFiles = ApiClient.convertToType(data['modelFiles'], [ModelFile]);
      if (data.hasOwnProperty('properties'))
        obj.properties = ApiClient.convertToType(data['properties'], {'String': Object});
    }
    return obj;
  }

  /**
   * The model, the endpoint is associated with.
   * @member {module:model/EntityReference} model
   */
  exports.prototype.model = undefined;

  /**
   * The model files, the endpoint requires.
   * @member {Array.<module:model/ModelFile>} modelFiles
   */
  exports.prototype.modelFiles = undefined;

  /**
   * The configuration of the endpoint.
   * @member {Object.<String, Object>} properties
   */
  exports.prototype.properties = undefined;


  return exports;

}));
