// speech_to_text_api_v30/src/model/ProjectProperties.js
/*
 * Speech to Text API v3.0
 * Speech to Text API v3.0.
 *
 * OpenAPI spec version: v3.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.21
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.SpeechToTextApiV30) {
      root.SpeechToTextApiV30 = {};
    }
    root.SpeechToTextApiV30.ProjectProperties = factory(root.SpeechToTextApiV30.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The ProjectProperties model module.
   * @module model/ProjectProperties
   * @version v3.0
   */

  /**
   * Constructs a new <code>ProjectProperties</code>.
   * @alias module:model/ProjectProperties
   * @class
   */
  var exports = function() {
  };

  /**
   * Constructs a <code>ProjectProperties</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/ProjectProperties} obj Optional instance to populate.
   * @return {module:model/ProjectProperties} The populated <code>ProjectProperties</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('datasetCount'))
        obj.datasetCount = ApiClient.convertToType(data['datasetCount'], 'Number');
      if (data.hasOwnProperty('evaluationCount'))
        obj.evaluationCount = ApiClient.convertToType(data['evaluationCount'], 'Number');
      if (data.hasOwnProperty('modelCount'))
        obj.modelCount = ApiClient.convertToType(data['modelCount'], 'Number');
      if (data.hasOwnProperty('transcriptionCount'))
        obj.transcriptionCount = ApiClient.convertToType(data['transcriptionCount'], 'Number');
      if (data.hasOwnProperty('endpointCount'))
        obj.endpointCount = ApiClient.convertToType(data['endpointCount'], 'Number');
    }
    return obj;
  }

  /**
   * The number of datasets associated to this project.
   * @member {Number} datasetCount
   */
  exports.prototype.datasetCount = undefined;

  /**
   * The number of evaluations associated to this project.
   * @member {Number} evaluationCount
   */
  exports.prototype.evaluationCount = undefined;

  /**
   * The number of models associated to this project.
   * @member {Number} modelCount
   */
  exports.prototype.modelCount = undefined;

  /**
   * The number of transcriptions associated to this project.
   * @member {Number} transcriptionCount
   */
  exports.prototype.transcriptionCount = undefined;

  /**
   * The number of endpoints associated to this project.
   * @member {Number} endpointCount
   */
  exports.prototype.endpointCount = undefined;


  return exports;

}));
