// speech_to_text_api_v30/src/model/ProjectLinks.js
/*
 * Speech to Text API v3.0
 * Speech to Text API v3.0.
 *
 * OpenAPI spec version: v3.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.21
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.SpeechToTextApiV30) {
      root.SpeechToTextApiV30 = {};
    }
    root.SpeechToTextApiV30.ProjectLinks = factory(root.SpeechToTextApiV30.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The ProjectLinks model module.
   * @module model/ProjectLinks
   * @version v3.0
   */

  /**
   * Constructs a new <code>ProjectLinks</code>.
   * @alias module:model/ProjectLinks
   * @class
   */
  var exports = function() {
  };

  /**
   * Constructs a <code>ProjectLinks</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/ProjectLinks} obj Optional instance to populate.
   * @return {module:model/ProjectLinks} The populated <code>ProjectLinks</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('evaluations'))
        obj.evaluations = ApiClient.convertToType(data['evaluations'], 'String');
      if (data.hasOwnProperty('datasets'))
        obj.datasets = ApiClient.convertToType(data['datasets'], 'String');
      if (data.hasOwnProperty('models'))
        obj.models = ApiClient.convertToType(data['models'], 'String');
      if (data.hasOwnProperty('endpoints'))
        obj.endpoints = ApiClient.convertToType(data['endpoints'], 'String');
      if (data.hasOwnProperty('transcriptions'))
        obj.transcriptions = ApiClient.convertToType(data['transcriptions'], 'String');
    }
    return obj;
  }

  /**
   * The location to get a list of all evaluations of this project.
   * @member {String} evaluations
   */
  exports.prototype.evaluations = undefined;

  /**
   * The location to get a list of all datasets of this project.
   * @member {String} datasets
   */
  exports.prototype.datasets = undefined;

  /**
   * The location to get a list of all models of this project.
   * @member {String} models
   */
  exports.prototype.models = undefined;

  /**
   * The location to get a list of all endpoints of this project.
   * @member {String} endpoints
   */
  exports.prototype.endpoints = undefined;

  /**
   * The location to get a list of all transcriptions of this project.
   * @member {String} transcriptions
   */
  exports.prototype.transcriptions = undefined;


  return exports;

}));
