// javascript-client/index.js
/*
 * Swagger Demo API Doc
 * This is a restful api document of Swagger Demo.
 *
 * OpenAPI spec version: 1.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.19
 *
 * Do not edit the class manually.
 *
 */

(function(factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/ModelAndView', 'model/User', 'model/View', 'api/BasicErrorControllerApi', 'api/UserControllerApi'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('./ApiClient'), require('./model/ModelAndView'), require('./model/User'), require('./model/View'), require('./api/BasicErrorControllerApi'), require('./api/UserControllerApi'));
  }
}(function(ApiClient, ModelAndView, User, View, BasicErrorControllerApi, UserControllerApi) {
  'use strict';

  /**
   * This_is_a_restful_api_document_of_Swagger_Demo_.<br>
   * The <code>index</code> module provides access to constructors for all the classes which comprise the public API.
   * <p>
   * An AMD (recommended!) or CommonJS application will generally do something equivalent to the following:
   * <pre>
   * var SwaggerDemoApiDoc = require('index'); // See note below*.
   * var xxxSvc = new SwaggerDemoApiDoc.XxxApi(); // Allocate the API class we're going to use.
   * var yyyModel = new SwaggerDemoApiDoc.Yyy(); // Construct a model instance.
   * yyyModel.someProperty = 'someValue';
   * ...
   * var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
   * ...
   * </pre>
   * <em>*NOTE: For a top-level AMD script, use require(['index'], function(){...})
   * and put the application logic within the callback function.</em>
   * </p>
   * <p>
   * A non-AMD browser application (discouraged) might do something like this:
   * <pre>
   * var xxxSvc = new SwaggerDemoApiDoc.XxxApi(); // Allocate the API class we're going to use.
   * var yyy = new SwaggerDemoApiDoc.Yyy(); // Construct a model instance.
   * yyyModel.someProperty = 'someValue';
   * ...
   * var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
   * ...
   * </pre>
   * </p>
   * @module index
   * @version 1.0
   */
  var exports = {
    /**
     * The ApiClient constructor.
     * @property {module:ApiClient}
     */
    ApiClient: ApiClient,
    /**
     * The ModelAndView model constructor.
     * @property {module:model/ModelAndView}
     */
    ModelAndView: ModelAndView,
    /**
     * The User model constructor.
     * @property {module:model/User}
     */
    User: User,
    /**
     * The View model constructor.
     * @property {module:model/View}
     */
    View: View,
    /**
     * The BasicErrorControllerApi service constructor.
     * @property {module:api/BasicErrorControllerApi}
     */
    BasicErrorControllerApi: BasicErrorControllerApi,
    /**
     * The UserControllerApi service constructor.
     * @property {module:api/UserControllerApi}
     */
    UserControllerApi: UserControllerApi
  };

  return exports;
}));
